src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -12805,9 +12961,22 @@ void Assembler::prefix(Address adr, Register reg, bool byteinst) {
       }
     }
   }
+  if (is_map1) emit_int8(0x0F);
+}
+
+void Assembler::prefix_rex2(Address adr, Register reg, bool byteinst, bool is_map1) {
+  int bits = is_map1 ? REX2BIT_M0 : 0;
+  bits |= get_base_prefix_bits(adr.base()->encoding());
```

Comment:
This also needs fix to:
bits |= get_base_prefix_bits(adr.base());

Reply:
Thank you @sviswa7, made the change.